### ISSUE TITLE
HFI Calculator: Convert fuel type cells to a select field / dropdown

### ISSUE THREAD

**Issue by tessmgood** [bcgov/wps#1555]
**As a** *(User Type/Persona)* **I want** *(Feature/enhancement)* **So That** *(Value, why is this wanted, what is the user trying to accomplish)*

**Additional Context**
- enter text here
- enter text here

**Acceptance Criteria**
- [ ] Given (Context), When (action carried out), Then (expected outcome)
- [ ] Given (Context), When (action carried out), Then (expected outcome)

**Definition of Done**
- [ ] Ready to Demo in Sprint Review
- [ ] Does what I have made have appropriate test coverage?
- [ ] Documentation and/or scientific documentation exists and can be found
- [ ] Peer Reviewed by 2 people on the team
- [ ] Manual testing of all PRs in Dev and Prod
- [ ] Merged

**Comment by conbrad** [comment]
Permission as this as well?

**Comment by Sybrand** [comment]
@conbrad re. permissions - are you seeing fine grained permissions? Right now I'm seeing "view" and "edit" roles. I'd assume anyone with an idir can view - so maybe that's not even a hard permission, but edit is then required to see it on the front end, and to post/put to the back-end.

**Comment by conbrad** [comment]
@Sybrand yeah I think all we're concerned about are those 2 permissions at the moment, which makes things easier to get started.

**Comment by conbrad** [comment]
- Given I change a fuel type for a weather station, that change persists for the user that made the change within browser sessions, pending #1634 decision

We should have requests persisted to database by the time we tackle this ticket, I think this AC can be updated to be persisted across all users instead of per user browser session.

**Comment by conbrad** [comment]
A lot of this work has been made more manageable with the current work we're doing migrating the logic from frontend to backend. It may be a good opportunity to refactor the request/response models for HFI in this ticket to be more explicit, since this is post MVP and we'll have the clearest picture of field structure when tackling this ticket.

Thoughts, @Sybrand ?

**Comment by andrea-williams** [comment]
re: RISK in Additional Context. I'm thinking we'll need to implement a lockfile mechanism for writing changes to the DB so that we don't encounter race conditions when 2 people try to save changes simultaneously. Not sure how to handle situation where the page is stale for a viewer because someone else has recently submitted changes...
Or maybe I'm over-complicating this.

**Comment by Sybrand** [comment]
@andrea-williams re. overwriting. It's more complicated than that (meaning - let's leave it for now) ; You'd need to push changes to all other open browsers to avoid overwrites. So we'd need a push mechanism - something we haven't thought about at all.
